src/webui/www/public/scripts/dynamicTable.js

```diff
@@ -489,22 +489,31 @@ var DynamicTable = new Class({
         },
 
         selectRow : function (rowId) {
-            this.selectedRows.empty();
+            this.deselectAll();
             this.selectedRows.push(rowId);
-            var trs = this.tableBody.getElements('tr');
-            for (var i = 0; i < trs.length; i++) {
-                var tr = trs[i];
-                if (tr.rowId == rowId) {
-                    if (!tr.hasClass('selected'))
-                        tr.addClass('selected');
-                }
+            this.tableBody.getElements('tr').each(function(tr) {
+                if (tr.rowId == rowId)
+                    tr.addClass('selected');
                 else
-                if (tr.hasClass('selected'))
                     tr.removeClass('selected');
-            }
+            });
             this.onSelectedRowChanged();
         },
 
+        reselectRows : function(rowIds) {
+            this.deselectAll();
+
+            var that = this;
+            rowIds.each(function(rowId) {
```

Comment:
just `that.selectedRows = rowIds`?

Reply:
I considered doing that but if there are any additional properties on the `selectedRows` array, they'll be lost when assigning `rowIds` to it. I don't know if that's purely a theoretical concern or if it's also practical.